### PR TITLE
Add documentation for primer and convert to command line option

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Based on https://docs.pytest.org/en/latest/example/simple.html#control-skipping-of-tests-according-to-command-line-option

Because we can remove the standard ``addopts`` IDEs like VS Code will now actually find all tests in our test suite, but will just skip them when running normally. Makes it easier (at least for those on VS Code) to explore all tests.

Ticks of two points from #5306

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -508,7 +508,7 @@ jobs:
         run: |
           . venv/bin/activate
           pip install -e .
-          pytest -m primer_stdlib -n auto
+          pytest --primer_stdlib
```
Comment (DanielNoord) [inline .github/workflows/ci.yaml:511]:
```suggestion
          pytest --primer_stdlib -n auto
```

File: tests/conftest.py [hunk 2]
```diff
@@ -56,3 +56,39 @@ def disable():
 @pytest.fixture(scope="module")
 def reporter():
     return MinimalTestReporter
+
+
+def pytest_addoption(parser) -> None:
+    parser.addoption(
+        "--primer_stdlib",
```
Comment (Pierre-Sassoulas) [inline tests/conftest.py:63]:
```suggestion
        "--primer-stdlib",
```

File: tests/conftest.py [hunk 3]
```diff
@@ -56,3 +56,39 @@ def disable():
...
+        action="store_true",
+        default=False,
+        help="Run primer stdlib tests",
+    )
+    parser.addoption(
+        "--primer_external",
```
Comment (Pierre-Sassoulas) [inline tests/conftest.py:69]:
```suggestion
        "--primer-external",
```
There's a lot of other place to change, but it's hard to do in github.

File: tests/conftest.py (hunk 2 above)
Comment (DanielNoord) [reply tests/conftest.py:63]: You can't mark with `@pytest.mark.primer_stdlib` so I thought I would keep consistency throughout the functions using it. Do you want me to change it anyway?

Comment (Pierre-Sassoulas) [reply tests/conftest.py:63]: For CLI option I think the standard is ``-``

File: doc/development_guide/testing.rst [hunk 4]
```diff
@@ -129,6 +129,19 @@ and should exit with exit code 2 the ``.out`` file should be named ``bad_configu
...
 
+Primer tests
+-------------------------------------------
+
+Pylint also uses what we refer to as ``primer`` tests. These are tests that are ran automatically
+in our Continuous Integration and check whether any changes in Pylint lead to crashes or fatal errors
+on the ``stdlib`` and a selection of external repositories.
+This list of repositories is selected on the basis of three criteria: 1) projects need to use a diverse
+range of language features, 2) projects need to be well maintained and 3) projects should not have a codebase
+that is too repeitive. This guarantees a good balance between speed of our CI and finding potential bugs.
```
Comment (jacobtylerwalls) [inline doc/development_guide/testing.rst:140]:
```suggestion
that is too repetitive. This guarantees a good balance between speed of our CI and finding potential bugs.
```

File: doc/development_guide/testing.rst [hunk 5]
```diff
@@ -129,6 +129,19 @@ and should exit with exit code 2 the ``.out`` file should be named ``bad_configu
 The content of the ``.out`` file should have a similar pattern as a normal Pylint output. Note that the
 module name should be ``{abspath}`` and the file name ``{relpath}``.
 
+Primer tests
+-------------------------------------------
+
+Pylint also uses what we refer to as ``primer`` tests. These are tests that are ran automatically
```
Comment (jacobtylerwalls) [inline doc/development_guide/testing.rst:135]:
```suggestion
Pylint also uses what we refer to as ``primer`` tests. These are tests that are run automatically
```

File: doc/development_guide/testing.rst (hunk 4 above)
Comment (DanielNoord) [reply doc/development_guide/testing.rst:140]: 😅

**Comment by DanielNoord** [comment]
We still need to add the mark to the `pytest` command as otherwise the CI job will run all tests including the `primer`. I think the latest commit fixes the increased run time of `primer` in this PR.

**Comment by Pierre-Sassoulas** [comment]
> We still need to add the mark to the pytest command as otherwise the CI job will run all tests including the primer

Wasn't the point of adopt was that primer were not launched by default ? What am I missing ?

**Comment by DanielNoord** [comment]
> We still need to add the mark to the `pytest` command as otherwise the CI job will run all tests including the `primer`. I think the latest commit fixes the increased run time of `primer` in this PR.

This has been fixed with https://github.com/PyCQA/pylint/pull/5387/commits/dab1e030f277d4a894802d5c826ad0ac62053415

> Wasn't the point of adopt that primer were not launched by default ? What am I missing ?

With this implementation ``--primer-external`` is ``False`` by default. It looks in the ``item.keywords`` (decorators) and if ``primer_stdlib`` is in there it adds a ``skip``. If the option is supplied this won't happen so the mark isn't skipped. We need to add `-m primer_external` to then tell it to only run the marked tests again.

Basically instead of not selecting the mark by default (and therefore not showing up in test collection) this makes it so the mark is skipped by default.
